nexusutils/nyaml2nxdl/nyaml2nxdl_helper.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+"""Main file of yaml2nxdl tool.
+Users create NeXus instances by writing a YAML file
+which details a hierarchy of data/metadata elements
+
+"""
+# -*- coding: utf-8 -*-
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+# Yaml library does not except the keys (escapechar "\t" and yaml separator ":")
+# So the corresponding value is to skip them and
+# and also carefull about this order
+ESCAPE_CHAR_DICT = {":": "\':\'",
+                    "\t": "    "}
```

Comment:
Does not the conversion  : -> ':' what Florian also mentioned cause any problems?

Reply:
@sanbrock This does not necessarily create problems it is just inconvenient and in my small tests it worked to use this in pyyaml. It would therefore be nice to not having to type this in the yaml files, but this can also be fixed at a later stage.

Reply:
Because of ':' yaml to nxdl giving error in file NXlog_parsed.yaml for line:

   `deprecated: see: https://github.com/nexusformat/definitions/issues/639`